src/xenia/memory.cc

```diff
@@ -353,6 +353,25 @@ BaseHeap* Memory::LookupHeapByType(bool physical, uint32_t page_size) {
 
 VirtualHeap* Memory::GetPhysicalHeap() { return &heaps_.physical; }
 
+void Memory::GetHeapsPageStatsSummary(std::vector<BaseHeap*> provided_heaps,
```

Comment:
The vector should not be passed as value as it causes unneeded allocations, it should be `const std::vector&` if it's read-only or `std::vector&` if writable. For read-only, what's even more preferred is a raw pointer to the first element + count. for greater flexibility